eval.ml

```diff
@@ -94,11 +95,19 @@ let value_eq (v1 : value) (v2 : value) =
   | Int n1, Int n2 -> n1 = n2
   | Int _, _ | _, Int _ -> false
   | Fun_ref f1, Fun_ref f2 -> f1 = f2
+  | Fun_ref _, _ | _, Fun_ref _ -> false
+  | Array vs1, Array vs2 ->
```

Comment:
Minor: could you add a case for `Array _, _ | _, Array _`? I know it's redundant (but some of the other patterns are already redundant), but it makes the case explicit and then you won't forget to add it if you add another value type.